### ISSUE TITLE
Flushing file

### ISSUE THREAD

**Issue by kiranshila** [georust/netcdf#135]
Hello again! Another quick issue/question - 

Is there an equivalent to `nc_sync` / flushing to disk in this rust implementation? I'm getting some errors occasionally that look like

```
HDF5.API.H5Error: Error reading dataset /voltages
libhdf5 Stacktrace:
  [1] H5FD_read: Invalid arguments to routine/Address overflowed
      addr overflow, addr = 16803176, size = 3656, eoa = 16803176
```

when trying to read from other programs. Looking around this seems to be the result of an incomplete write.

**Comment by mulimoen** [comment]
We should expose `nc_sync` to allow the user to sync the file. In the mean time you could use [`open_with`](https://docs.rs/netcdf/latest/netcdf/fn.open_with.html) with the [`SHARE`](https://docs.rs/netcdf/latest/netcdf/struct.Options.html#associatedconstant.SHARE) flag.

**Comment by lnicola** [comment]
Can you get the native handle to call `nc_sync` on it? I couldn't spot it in the docs.

**Comment by mulimoen** [comment]
No, the native handle is kept private

**Comment by mulimoen** [comment]
It should be mentioned that reading a file whilst writing to it is not supported by hdf5 unless SWMR mode is set. There is a PR implementing this in `netcdf-c` here: https://github.com/Unidata/netcdf-c/pull/2653

**Comment by mulimoen** [comment]
@kiranshila This is published in `0.9.1` and can be called as [`FileMut::sync`](https://docs.rs/netcdf/latest/netcdf/struct.FileMut.html#method.sync).
